src/precice/impl/SolverInterfaceImpl.cpp

```diff
@@ -1258,6 +1258,7 @@ void SolverInterfaceImpl::readBlockScalarData(
                 data.getName());
   auto &     valuesInternal = data.values();
   const auto vertexCount    = valuesInternal.size();
+  // Open issue: how to handle filtered data sets? valuesInternal is set to zero initially
```

Comment:
Let's not merge this like this. Will get forgotten. In my opinion the current solution is OK enough. Better open an issue if you want to revisit this in the future.